src/main/java/ch/njol/skript/doc/JSONGenerator.java

```diff
@@ -60,47 +66,91 @@ public JSONGenerator(File templateDir, File outputDir) {
 		if (nameAnnotation == null || syntaxClass.getAnnotation(NoDoc.class) != null)
 			return null;
 		JsonObject syntaxJsonObject = new JsonObject();
+
 		syntaxJsonObject.addProperty("id", DocumentationIdProvider.getId(syntaxInfo));
 		syntaxJsonObject.addProperty("name", nameAnnotation.value());
+		syntaxJsonObject.add("patterns", convertToJsonArray(syntaxInfo.getPatterns()));
 
 		Since sinceAnnotation = syntaxClass.getAnnotation(Since.class);
-		syntaxJsonObject.add("since", sinceAnnotation == null ? null : convertToJsonArray(sinceAnnotation.value()));
+		syntaxJsonObject.add("since", convertToJsonArray(sinceAnnotation.value()));
 
 		Description descriptionAnnotation = syntaxClass.getAnnotation(Description.class);
-		if (descriptionAnnotation != null) {
-			syntaxJsonObject.add("description", convertToJsonArray(descriptionAnnotation.value()));
-		} else {
-			syntaxJsonObject.add("description", new JsonArray());
-		}
+		syntaxJsonObject.add("description", convertToJsonArray(descriptionAnnotation.value()));
 
 		Examples examplesAnnotation = syntaxClass.getAnnotation(Examples.class);
-		if (examplesAnnotation != null) {
-			syntaxJsonObject.add("examples", convertToJsonArray(examplesAnnotation.value()));
-		} else {
-			syntaxJsonObject.add("examples", new JsonArray());
-		}
+		syntaxJsonObject.add("examples", convertToJsonArray(examplesAnnotation.value()));
```

Comment:
any reason why this null checking is being removed?

Reply:
in an earlier review comment, we had decided to use null when applicable

Reply:
sovde's comment is valid, but ill make convertToJsonArray not return `null`, instead returning an empty array if the param is null

Reply:
I'd prefer to have null, I think that makes more sense as the underlying value was null.

Reply:
null arrays aren't serialized though, so they'll still be missing from the file

Reply:
ooh good call, it seems you can call serializeNulls when creating the gson instance to allow this

Reply:
should be good now